### PR TITLE
Add getters for time

### PR DESCRIPTION
**Pull request by simonvandel**
Adds bindings for the following options.
- CURLINFO_TOTAL_TIME
- CURLINFO_NAMELOOKUP_TIME
- CURLINFO_CONNECT_TIME
- CURLINFO_APPCONNECT_TIME
- CURLINFO_PRETRANSFER_TIME
- CURLINFO_STARTTRANSFER_TIME
- CURLINFO_REDIRECT_TIME

Is there a better way than transmute in getopt_double?

### CODE REVIEW

File: src/easy.rs [hunk 1]
```diff
@@ -2864,6 +2951,11 @@ impl Easy {
         }
     }
 
+    fn getopt_double(&mut self, opt: curl_sys::CURLINFO) -> Result<c_double, Error> {
+        self.getopt_ptr(opt)
+            .map(|p| unsafe {transmute::<_, c_double>(p)})
```
Comment (alexcrichton) [inline src/easy.rs]: This looks to be failing on 32-bit platforms, perhaps this could just call the FFI fn directly?

Comment (simonvandel) [inline src/easy.rs]: I am not really sure what you mean by calling the FFI function directly.
How does one go from *const i8 to f64 in a portable way?

**Comment by simonvandel** [comment]
I agree on Duration being more rustic, implemented in newest commit.

**Comment by simonvandel** [comment]
I added a `as u64` cast on the pointer which at least makes it compile. I haven't tested whether the returned value is sensible on 32bit though!

File: src/easy.rs [hunk 2]
```diff
@@ -2864,6 +2958,11 @@ impl Easy {
         }
     }
 
+    fn getopt_double(&mut self, opt: curl_sys::CURLINFO) -> Result<c_double, Error> {
+        self.getopt_ptr(opt)
+            .map(|p| unsafe {transmute::<_, c_double>(p as u64)})
```
Comment (alexcrichton) [inline src/easy.rs]: I think we'll still want to call the raw FFI function directly here b/c we want 64-bits of information back, but on 32-bit this is only giving us 32-bits of information back

**Comment by simonvandel** [comment]
The failing test on AppVeyor should be unrelated. Do you want me to squash the two commits?

**Comment by alexcrichton** [comment]
Nah looks good to me, thanks!

**Comment by simonvandel** [comment]
I'm working on a project using tokio_curl and this library. Could you update the crates.io versions of these libraries so I can remove my local forks?

**Comment by alexcrichton** [comment]
Certainly!

**Pull request merged**